persim/visuals.py

```diff
@@ -280,3 +280,178 @@ def wasserstein_matching(I1, I2, matchidx, palette=None, labels=["dgm1", "dgm2"]
+
+        if export:
+            fp = io.BytesIO()
+            plt.savefig(fp, dpi=dpi)
+            fp.seek(0)
+
+            out += [fp]
+
+        if show:
+            plt.show()
+
+        return out
+
+    def _plot_Hn(self, **kwargs):
+        '''
+        parameters:
+        ===========
+            figsize: tuple
+                figure size, default=(6,6)
+
+            show: boolean
+                show the figure via plt.show()
+
+            export_png: boolean
+                write image to png data, returned as io.BytesIO() instance, default=False
+                
+            **kwargs: artist paramters for the barcodes, defaults:
+                c='grey'
+                linestyle='-'
+                linewidth=0.5
+                dpi=100 (for png export)
+
+        returns:
+        ===========
+            list of png exports or []
+        '''
+        import io
+        
+        fsize = kwargs.get('figsize', (6, 4))
+        show = kwargs.get('show', True)
+        export = kwargs.get('export_png', False)
+        dpi = kwargs.get('dpi', 100)
+
+        out = []
+
+        for dim, diagram in enumerate(self.diagrams):
+            fig, ax = plt.subplots(1, 1, figsize=fsize)
+
+            self._plot_many_bars(dim, diagram, 0, [ax], **kwargs)
+
+            if export:
+                fp = io.BytesIO()
+                plt.savefig(fp, dpi=dpi)
+                fp.seek(0)
+
+                out += [fp]
+
+            if show:
+                plt.show()
+
+        return out
+
+    def _plot_many_bars(self, dim, diagram, idx, ax, **kwargs):
+        number_of_bars = len(diagram)
+        print("Number of bars in dimension %d: %d" % (dim, number_of_bars))
+
+        if number_of_bars > 0:
+            births = np.vstack([(elem[0], i) for i, elem in enumerate(diagram)])
+            deaths = np.vstack([(elem[1], i) for i, elem in enumerate(diagram)])
+
+            inf_bars = np.where(np.isinf(deaths))[0]
+            max_death = deaths[np.isfinite(deaths[:, 0]), 0].max()
+
+            number_of_bars_fin = births.shape[0] - inf_bars.shape[0]
+            number_of_bars_inf = inf_bars.shape[0]
+
+            _ = [self._plot_a_bar(ax[idx], birth, deaths[i], max_death, **kwargs) for i, birth in enumerate(births)]
+
+        # the line below is to plot a vertical red line showing the maximal finite bar length
+        ax[idx].plot([max_death, max_death], [0, number_of_bars - 1],
+            c='r',
+            linestyle='--',
+            linewidth=0.5)
+
+        title = "H%d barcode: %d finite, %d infinite" % (dim, number_of_bars_fin, number_of_bars_inf)
+        ax[idx].set_title(title, fontsize=9)
+        ax[idx].set_yticks([])
+
+        ax[idx].spines['right'].set_visible(False)
+        ax[idx].spines['left'].set_visible(False)
+        ax[idx].spines['top'].set_visible(False)
+
+    @staticmethod
+    def _plot_a_bar(ax, birth, death, max_death, c='gray', linestyle='-', linewidth=0.5, **kwargs):
```

Comment:
is kwargs used at all? Is it necessary here?

Reply:
Possibly; if the user passes addition plotting parameters that would get sent to `ax.plot()` then this `**kwargs` would catch it. At least that is what is supposed to happen, they were actually not being consumed. Oops!